File: Java/_10799.java
package Algorithm.BaekJoon;

import java.util.*;
import java.io.*;

// 스택
public class _10799 {

    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String line = br.readLine();

        System.out.println(cntBar(line));
    }

    static int cntBar(String line){

        int result = 0;

        Stack<Integer> stack = new Stack<>();
        for(int i = 0; i < line.length(); i++){
            if(line.charAt(i) == '('){
                stack.push(i);
            } else if(line.charAt(i) == ')'){
                if(stack.peek() == i-1){
                    stack.pop();
                    result += stack.size();
                } else{
                    stack.pop();
                    result += 1;
                }
            }
        }
        return result;
    }
}

File: Python/10799.py
import sys

def cntBar(line):
    result = 0

    stack = list()
    for i, t in enumerate(line):
        if t == '(':
            stack.append(i)
        elif t == ')':
            if stack[-1] == i-1:
                stack.pop()
                result += len(stack)
            else:
                stack.pop()
                result += 1

    return result;

f = lambda : sys.stdin.readline().rstrip()

line = f()

print(cntBar(line))